auth-service/src/main/java/com/petclinic/auth/User/UserServiceImpl.java

```diff
@@ -17,27 +15,49 @@
 public class UserServiceImpl implements UserService{
 
     private final UserRepo userRepo;
+
     private final UserMapper userMapper;
 
+    @Override
+    public User findUserById(long id) {
+        User entity  = userRepo.findById(id)
+                .orElseThrow(() -> new NotFoundException("No user found for userID" + id));
+        log.info("User getUserById: found userId: {}", entity.getId());
+        return entity;
+    }
+
+    @Override
+    public Page<User> findAll(PageRequest of) {
+        return userRepo.findAll(of);
+    }
+
     @Override
     public User createUser(@Valid UserIDLessDTO userIDLessDTO) {
 
-        if (userRepo.findByEmail(userIDLessDTO.getEmail()) != null){
-            throw new DuplicateKeyException("Duplicate email for " + userIDLessDTO.getEmail());
-        }
-        else {
-            log.info("Saving user with username {}", userIDLessDTO.getUsername());
+        try {
+            log.info("Saving user with email {}", userIDLessDTO.getEmail());
             User user = userMapper.idLessDTOToModel(userIDLessDTO);
             return userRepo.save(user);
         }
+        catch (DuplicateKeyException e){
+            throw new DuplicateKeyException("Duplicate email for userEmail " + userIDLessDTO.getEmail());
```

Comment:
Why catch and re-throw the same exception with a different message?
Wouldn't it be better to just let it be thrown as is and add a `@Throws` clause?